sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/optimizer/Optimizer.scala

```diff
@@ -199,6 +199,8 @@ abstract class Optimizer(catalogManager: CatalogManager)
     // idempotence enforcement on this batch. We thus make it FixedPoint(1) instead of Once.
     Batch("Join Reorder", FixedPoint(1),
       CostBasedJoinReorder) :+
+    Batch("Reorder Predicate", FixedPoint(1),
```

Comment:
Does `Once` fail any tests? It's best to make batches idempotent - it's a helpful test feature.

Reply:
Thank you @tanelk, change it to Once.